common/test/acceptance/tests/studio/test_studio_course_team.py

```diff
@@ -352,3 +356,46 @@ def test_admins_should_be_able_to_give_course_ownership_to_someone_else(self):
 
         self.log_in(self.user)
         self._assert_current_course(visible=False)
+
+    def test_third_party_auth_on_sudo_page_with_unlinked_account(self):
+        """
+        Test that dummy auth button is disabled on sudo page when no account is linked.
+        """
+        # Logout and log back in to remove sudo access.
+        LogoutPage(self.browser).visit()
+        self.log_in(user=self.user)
+
+        sudo_password_page = SudoPage(self.browser, self.page)
+        sudo_password_page.visit()
+        self.assertTrue(sudo_password_page.is_dummy_auth_button_disabled)
+
+    def test_third_party_auth_on_sudo_page_with_linked_account(self):
+        """
+        Test that user can authenticate on sudo page with dummy third party auth.
+        """
+        dummy_user = self._make_user(username='dummy_user')
+
+        # Add user to course team.
+        self.page.visit()
+        self.page.add_user_to_course(dummy_user['email'])
+        self.assertIn(dummy_user.get('username'), self.page.usernames)
+
+        # Logout and log back in with new user.
+        LogoutPage(self.browser).visit()
+        self.log_in(user=dummy_user)
+
+        self._link_dummy_account()
+
+        # Visit sudo page and click on dummy auth button to get sudo access.
+        sudo_password_page = SudoPage(self.browser, self.page)
+        sudo_password_page.visit()
+        sudo_password_page.click_third_party_dummy_provider_button()
```

Comment:
I now see that the assertion is hidden in the `wait_for_page()` call in `click_third_party_dummy_provider_button()`. I think that assertions belongs here rather than buried in a PageObject method.